packages/monocdk-experiment/gen.js

```diff
@@ -0,0 +1,99 @@
+// generates the mono-cdk module by copying @aws-cdk/lib/** to src/
+// and rewriting the interdepedent "import" statements.
+const fs = require('fs-extra');
+const path = require('path');
+const glob = require('glob');
+
+async function main() {
+  const srcdir = path.resolve('src');
+  const reexports = [];
+
+  await fs.remove(srcdir);
+  await fs.mkdir(srcdir);
+
+  const root = path.resolve(__dirname, '..', '@aws-cdk');
+  const modules = await fs.readdir(root);
+
+  for (const dir of modules) {
+    const moduledir = path.resolve(root, dir);
+    const meta = JSON.parse(await fs.readFile(path.join(moduledir, 'package.json'), 'utf-8'));
+
+    if (meta.deprecated) {
+      console.error(`skipping deprecated ${meta.name}`);
+      continue;
+    }
+
+    if (!meta.jsii) {
+      console.error(`skipping non-jsii ${meta.name}`);
+      continue;
+    }
+
+    // check if moduledir includes any directory other than "lib" and "test"
+    const subdirs = [];
+    const allowed = [ 'lib', 'test', 'node_modules', 'scripts', 'build-tools' ];
```

Comment:
This could be a `Set` that you'd move outside of the loop...

Reply:
I think that's fine for now